src/main/java/com/example/staffmanagerapi/handler/GlobalExceptionHandler.java

```diff
@@ -13,52 +17,90 @@
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
 
-import java.util.*;
-
 @ControllerAdvice
 public class GlobalExceptionHandler {
 
-    @ExceptionHandler(EntityNotFoundException.class)
-    public ResponseEntity<Object> handleEntityNotFoundException(
-            EntityNotFoundException ex
-    ) {
-        return ResponseEntity
-                .status(HttpStatus.NOT_ACCEPTABLE)
-                .body(ex.getMessage());
-    }
-
-    @ExceptionHandler(MethodArgumentNotValidException.class)
-    public ResponseEntity<Object> handleException(
-            MethodArgumentNotValidException ex
-    ) {
-        BindingResult bindingResult = ex.getBindingResult();
-        List<FieldError> fieldErrors = bindingResult.getFieldErrors();
-        Map<String, String> errorMessages = new HashMap<>();
+  @ExceptionHandler(EntityNotFoundException.class)
+  public ResponseEntity<ResponseTemplate> handleEntityNotFoundException(
+    EntityNotFoundException ex
+  ) {
+    return ResponseEntity
+      .status(HttpStatus.NOT_ACCEPTABLE)
```

Comment:
je comprends pas le `NOT_ACCEPTABLE` 🤔

Normalement on reçoi un 406 si on fait appel a une ressource avec des requests headers non valide, par exemple on met `Content-Type: application/xml`, alors que sa retourne du JSON.

Entity not found, la c'est un 404 non ?

Reply:
Pour être honnête j'ai mis les status code un peu a l'arrache, on peut créer une user story spécialement dédié a ça si necessaire